### ISSUE TITLE
Enhance user management functionality

### ISSUE THREAD

**Issue by willrennie** [salesagility/SuiteCRM#3277]
#### Issue
User management is restrictive especially for a CRM where there are a large number of users, in one example 500+

#### Expected Behavior
The requirement is to mass update all user values, mass set/update dashboards/dashlets, mass delete users, import users with pre-defined values and preferences etc.

#### Actual Behavior
Functionality for mass user management is lacking. Dashboard tabs and dashlets cannot be pre-defined for users, mass resets(reset homepage, reset password etc.) are not currently possible, mass deleting is not possible. User duplication also does not copy over all user attributes, such as Security groups and role relationships and data.

#### Possible Fix
Implement more comprehensive administrative functionality for managing CRM users.

#### Steps to Reproduce
No bug/steps to reproduce as this is a suggestion.

#### Context
The end goal is to obtain a more comprehensive set of administrative tools to manage CRM users on a daily basis for larger CRM deployments.

This should be a low priority as it is a suggestion

#### Your Environment
* SuiteCRM Version used: 7.7.9
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): All compatible browsers
* Environment name and version (e.g. MySQL, PHP 7): PHP 7, MySQL 5.6
* Operating System and version (e.g Ubuntu 16.04): Ubuntu 16.04

**Comment by gunnicom** [comment]
I would split dashboard management and user management into two issues.

**Comment by chris001** [comment]
Sounds like a perfect use case for a command line admin tool, `scc` = SuiteCRM Commandline, built on other open source command line admin tools.

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Enhance user management functionality](https://trello.com/c/vDTf349X/15-enhance-user-management-functionality)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
